src/ClusterManager/job_launcher.py

```diff
@@ -369,17 +369,34 @@ def status(self, refresh=False):
         phase = self.pod.status.phase
 
         # !!! Pod is running, doesn't mean "Role" is ready and running.
-        if(phase == "Running"):
+        if phase == "Running":
             # Found that phase won't turn into "Unkonwn" even when we get 'unknown' from kubectl
             if self.pod.status.reason == "NodeLost":
                 return "Unknown"
 
+            # Starting from v1.13, TaintBasedEvictions are enabled by default. NodeLost no longer
+            # exists. Use deletionTimstamp to signal node lost.
+            # See below for details:
+            # https://github.com/kubernetes/kubernetes/issues/72226
+            # https://kubernetes.io/docs/concepts/configuration/taint-and-toleration/
+            if self.pod.metadata.deletion_timestamp is not None:
+                logger.info("pod %s has deletion_timestamp %s. Marking pod as Unknown." %
+                            (self.pod_name, self.pod.metadata.deletion_timestamp))
+                return "Unknown"
+
             # Check if the user command had been ran.
             if not self._is_role_ready():
                 return "Pending"
 
         return phase
 
+    def pod_restricted_details(self):
+        detail = dict()
```

Comment:
Use literal value over constructor function: https://doughellmann.com/blog/2012/11/12/the-performance-impact-of-using-dict-instead-of-in-cpython-2-7-2/

Reply:
Nice.